sql/catalog/lease: TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces failed

sql/catalog/lease.TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12641444?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12641444?buildTab=artifacts#/) on master @ [51215136d2afb2c71eca999168071465cbcda15f](https://github.com/cockroachdb/cockroach/commits/51215136d2afb2c71eca999168071465cbcda15f):


```
=== RUN   TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/ce8d73b4a4081f39d36a7fa2c2508fcc/logTestRangefeedUpdatesHandledProperlyInTheFaceOfRaces2437494539
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:159: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:2340: test took 30.151976478s which is greater than 30s
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ce8d73b4a4081f39d36a7fa2c2508fcc/logTestRangefeedUpdatesHandledProperlyInTheFaceOfRaces2437494539
--- FAIL: TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces (30.17s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33462

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -2347,6 +2347,7 @@ func TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	defer ensureTestTakesLessThan(t, 30*time.Second)()
+	skip.UnderRace(t, "test must take less than 30 seconds, and race builds are slow")
 
 	ctx := context.Background()
 	var interestingTable atomic.Value

